source/concepts/naming-conventions.md

```diff
@@ -1,120 +1,117 @@
-Ember.js uses naming conventions to wire up your objects without a
-lot of boilerplate. You will want to use these conventional names
-for your routes, controllers and templates.
+Ember.js uses a runtime resolver to wire up your objects without a
+lot of boilerplate. As a developer, this resolver will work automatically
+if you place code in conventional locations within your project.
 
-You can usually guess the names, but this guide outlines, in one place, 
-all of the naming conventions. In the following examples 'App' is a name 
-that we chose to namespace or represent our Ember application when it was 
-created, but you can choose any name you want for your application.
-We will show you later how to create an Ember application, but for now we
-will focus on conventions.
+You can usually guess the names and locations, but this guide outlines, in one place,
+all of the naming conventions.
 
 ## The Application
 
-When your application boots, Ember will look for these objects:
+When your application boots, Ember will look for the objects exported
+by these modules in your project:
 
-* `App.ApplicationRoute`
-* `App.ApplicationController`
-* the `application` template
+* `app/app.js`
+* `app/controllers/application.js`
+* `app/templates/application.hbs`
 
 Ember.js will render the `application` template as the main template.
-If `App.ApplicationController` is provided, Ember.js will set an
-instance of `App.ApplicationController` as the controller for the
+If `controller:application` is provided, Ember.js will set an
+instance of `controller:application` as the controller for the
 template. This means that the template will get its properties from
 the controller.
 
-If your app provides an `App.ApplicationRoute`, Ember.js will invoke
+If your app provides an route at `app/routes/application.js` Ember.js will invoke
 [the][1] [router's][2] [hooks][3] first, before rendering the
-`application` template. Hooks are implemented as methods and provide 
-you access points within an Ember object's lifecycle to intercept and 
-execute code to modify the default behavior at these points to meet 
+`application` template. Hooks are implemented as methods and provide
+you access points within an Ember object's lifecycle to intercept and
+execute code to modify the default behavior at these points to meet
 your needs. Ember provides several hooks for you to utilize for various
-purposes (e.g. `model`, `setupController`, etc). In the example below 
-`App.ApplicationRoute`, which is an `Ember.Route` object, implements 
+purposes (e.g. `model`, `setupController`, etc). In the example below
+`App.ApplicationRoute`, which is an `Ember.Route` object, implements
```

Comment:
In some scenarios we talk in `type:name`, but here we reference the global location (`App.ApplicationRoute`).  Does this need to be changed?